extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -0,0 +1,295 @@
+# Define a Model Parser for LLama-Guard
+from typing import TYPE_CHECKING, Dict, List, Optional, Any
+
+import google.generativeai as genai
+
+
+from aiconfig.default_parsers.parameterized_model_parser import ParameterizedModelParser
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import ExecuteResult, Output, Prompt
+from aiconfig.util.params import resolve_prompt
+from aiconfig import CallbackEvent, get_api_key_from_environment, AIConfigRuntime
+
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from google.generativeai.types import AsyncGenerateContentResponse
+
+def construct_regular_outputs(response: "AsyncGenerateContentResponse") -> list[Output]:
+    """
+    Construct regular output per response result, without streaming enabled
+    """
+    output_list = []
+    for i, candidate in enumerate(response.candidates):
+        output = ExecuteResult(
+            **{
+                "output_type": "execute_result",
+                "data": candidate.content.parts[0].text,
+                "execution_count": i,
+                "metadata": response.prompt_feedback.__dict__,
```

Comment:
Where is prompt_feedback defined for Gemini? I've been looking at https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini#gemini-pro for the req/res but does `genai` sdk add this data?

Reply:
The Python sdk Docs were pretty lacking when building this, it was pretty inconsistent throughout their api docs and some of their guides didn't work. Also note that I mainily used their public api docs (not through vertex). Link: https://ai.google.dev/docs

The response object had a couple of attributes but it didn't make sense how to use all of them. Printing the object didn't give me any information either so I inspected it by doing `print(response.__dict__)` which shows me the object fields. Accessing fields like 

<img width="812" alt="Screenshot 2023-12-14 at 5 22 55 PM" src="https://github.com/lastmile-ai/aiconfig/assets/141073967/ea85a42a-fefa-453e-966f-c3399f62b77c">